source/NVDAObjects/window/winword.py

```diff
@@ -343,6 +343,30 @@
 	u'\uF0FC' : u'\u2714', # Wingdings
 }
 
+class ScreenUpdatingDisabler(object):

```

Comment:
While I understand the idea behind this, it doesn't look very pythonic to me. Is there any chance this can be implemented like a context manager?

Reply:
Not really, as there isn't actually a common point at which we could use a with block to control the lifetime of the context manager. Its lifetime affects anything between the time it was created and the end of the core cycle. I guess we could possibly override caretMovementScriptHelper and wrap the super call in a with statement. But there may be other situations other than caret movement where this could be used. Right now it is placed above each nvdahelper call, which ensures that if it is not disabled already, it will be now.

Reply:
Agreed that in that case, the current solution might be best. I think we must make sure that it doesn't cause any visual issues, though, since screen updating might now be off longer than before.